src/client.rs

```diff
@@ -268,33 +266,31 @@ where
         for i in (self.tuf.root().version() + 1)..latest_version {
             let version = MetadataVersion::Number(i);
 
-            let signed_root = self.remote.fetch_metadata(
+            let signed_root = await!(self.remote.fetch_metadata(
                 &root_path,
                 &version,
                 &self.config.max_root_size,
                 self.config.min_bytes_per_second,
                 None,
-            )?;
+            ))?;
 
             if !self.tuf.update_root(signed_root.clone())? {
                 error!("{}", err_msg);
                 return Err(Error::Programming(err_msg.into()));
             }
 
-            self.store_metadata(&root_path, &version, &signed_root);
+            await!(self.store_metadata(&root_path, &version, &signed_root));
         }
 
         if !self.tuf.update_root(latest_root.clone())? {
             error!("{}", err_msg);
             return Err(Error::Programming(err_msg.into()));
         }
 
-        self.store_metadata(
-            &root_path,
-            &MetadataVersion::Number(latest_version),
-            &latest_root,
-        );
-        self.store_metadata(&root_path, &MetadataVersion::None, &latest_root);
+        let latest_version = MetadataVersion::Number(latest_version);
+
+        await!(self.store_metadata(&root_path, &latest_version, &latest_root,));
```

Comment:
ditto RE "can these happen concurrently"?

Reply:
These however could be in parallel, though in practice this will just be writing to the local filesystem. Is there much advantage of still making this parallel?